src/appleseed/renderer/kernel/lighting/directlightingintegrator.cpp

```diff
@@ -670,13 +676,32 @@ void DirectLightingIntegrator::add_non_physical_light_sample_contribution(
     // Evaluate the light.
     Vector3d emission_position, emission_direction;
     Spectrum light_value(Spectrum::Illuminance);
-    light->evaluate(
+
+    // Generate a uniform sample in [0,1).
+    sampling_context.split_in_place(2, 1);
+    const Vector2d s = sampling_context.next2<Vector2d>();
+    float probability;
```

Comment:
The principle is the same as for mesh lights:

* Given a light emitting mesh, we first choose a triangle of the mesh with probability `Ptri`. Then, on this triangle, we choose a point with uniform probability `Ppoint` which is simply `1/triangleArea`. The total probability for that point is then `Ptri * Ppoint`.

* Given a set of non-physical lights, we first choose one of the light with probability `Plight` (in this method, `Plight` is `sample.m_probability`). Then, for this light, we choose a direction with probability `Pdir`. For most lights, given a target point to be illuminated, `Pdir` is just 1 since there isn't any choice to be made (given a point light and a target point, there is only one ray joining the two, hence its probability is 1). Similarly as for the mesh light case, the total probability for that light direction is then `Plight * Pdir`.